backend/services/implementations/testService.ts

```diff
@@ -253,20 +310,81 @@ class TestService implements ITestService {
     );
   }
 
+  private getQuestionComponents(
```

Comment:
i've spent a long time thinking about how to clean up these functions but i've given up 😢. lmk if you have ideas!